torch/ao/pruning/_experimental/pruner/base_structured_sparsifier.py

```diff
@@ -99,6 +103,10 @@ def _get_default_structured_pruning_patterns() -> Dict[
         # conv2d -> conv2d
         (nn.Conv2d, "output"): prune_conv2d,
         (nn.Conv2d, nn.Conv2d): prune_conv2d_conv2d,
+        # lstm -> getitem -> linear
+        (nn.LSTM, getitem, nn.Linear): prune_lstm_linear,
```

Comment:
I remember the output of LSTM is `output, (h_n, c_n)`, so there will be multiple getitem node after the call to forward, it won't affect the current use case much probably since getitem is followed by some other ops, but this is not very robust I think

Reply:
Yeah, this is very brittle - I tried to match `getitem(0)` instead of `getitem` only but couldn't figure out how to do that, if you have know how?

I can rename this to `prune_lstm_output_linear` I think to make it a bit more clear it's only for lstm -> output -> linear and not lstm -> h_n / c_n -> linear. The pruning function implementation is only correct lstm-> output -> linear anyways, it applies the wrong mask for h_n / c_n.